docs/en/3-Pipelines/Kubeflow-Pipelines.md

```diff
@@ -230,20 +247,478 @@ run = client.run_pipeline(
+during development, as well as for simple tasks that can be written and managed
+easily.
+
+This is an example of a lightweight pipeline with a single component that
+concatenates strings:
+
+```python
+import kfp
+from kfp import dsl
+from kfp.components import func_to_container_op
+
+def concat_string(a, b) -> str:
+    return f"({a} | {b})"
+
+concat_string_component = func_to_container_op(concat_string,
+                                               base_image="python:3.8.3-buster"
+                                               )
+@dsl.pipeline(
+    name="My lightweight pipeline",
+)
+def pipeline(str1, str2, str3):
+    # Note that we use the concat_string_component, not the
+    # original concat_string() function
+    concat_result_1 = concat_string_component(str1, str2)
+
+    # By using cancat_result_1's output, we define the dependency of
+    # concat_result_2 on concat_result_1
+    concat_result_2 = concat_string_component(concat_result_1.output, str3)
+```
+
+where we see our `concat_string` component is defined directly in python rather
+than from a Docker image. In the end our function still runs in a container, but
+instead of us building a new image the `func_to_container_op()` runs our python
+code inside the provided base image (`python:3.8.3-buster`). This lets us avoid
+authoring a new image each iteration. The base image can be anything accessible
+by Kubeflow, which includes all images in the Azure Container Registry and any
+[whitelisted images](https://github.com/StatCan/gatekeeper-policies/blob/master/general/container-allowed-images/constraint.yaml)
+from Docker hub.
+
+<!-- prettier-ignore -->
+??? danger "Lightweight components have a number of advantages but also some drawbacks"
+    See [this description](https://github.com/StatCan/jupyter-notebooks/blob/master/kfp-basics/demo_kfp_lightweight_components.ipynb) of their basic characteristics, as well as [this example](https://github.com/StatCan/jupyter-notebooks/blob/master/mapreduce-pipeline/Compute-Pi-with-lightweight-components-and-minio.ipynb) which uses them in a more complex pipeline
+
+<!-- prettier-ignore -->
+??? info "A convenient base image to use is the the image your notebook server is running"
+    By using the same image as your notebook server, you ensure Kubeflow Pipelines has the same packages available to it as the notebook you author from.  This can help avoid accidentally importing packages you have locally but not in Kubeflow.  You can find that link from the notebook server page as shown below, but make sure you prepend the registry URL (so the below image would have `base_image=k8scc01covidacr.azurecr.io/machine-learning-notebook-cpu:562fa4a2899eeb9ae345c51c2491447ec31a87d7`).  Note though that while using a fully featured base image for iteration is fine, for production pipelines it is good to keep these images lean and only supply the necessary software.  That way you reduce the startup time for each step in your pipeline.
```

Comment:
```suggestion
    By using the same image as your notebook server, you ensure Kubeflow Pipelines has the same packages available to it as the notebook where you do your analysis. This can help avoid errors from importing packages specific specific to your environment. You can find that link from the notebook server page as shown below, but make sure you prepend the registry URL (so the below image would have `base_image=k8scc01covidacr.azurecr.io/machine-learning-notebook-cpu:562fa4a2899eeb9ae345c51c2491447ec31a87d7`). Note that while using a fully featured base image for iteration is fine, it's a good practice to keep production pipelines lean and only supply the necessary software. That way you reduce the startup time for each step in your pipeline.
```

And line length.